File: Card/Card.cpp
#include <iostream>

using std::cout;
using std::endl;

enum Suit { CLUBS, SPADES, HEARTS, DIAMONDS };

class Card {
  public:
    Card(int r, Suit s);
    virtual int value();
    Suit suit();
  protected:
    int m_card;
  private:
    Suit m_suit;
};

Card::Card(int r, Suit s) :m_card(r), m_suit(s) {}

int Card::value() {
  return m_card;
}

Suit Card::suit() {
  return m_suit;
}

class BlackJackCard : public Card {
  public:
    BlackJackCard(int r, Suit s);
    int value();
  private:
/* deliberately empty */
};

BlackJackCard::BlackJackCard(int r, Suit s) : Card(r, s) {}

int BlackJackCard::value() {
  if (m_card == 1) {
    return 11;
  } else if (m_card < 10) {
    return m_card;
  } else {
    return 10;
  }
}

int main() {
  Card c(5, CLUBS);
  cout << "c is a " << c.value() << " of " << c.suit() << endl;
  BlackJackCard j(12, HEARTS);
  cout << "j is a " << j.value() << " of " << j.suit() << endl;
  BlackJackCard k(1, HEARTS);
  cout << "k is a " << k.value() << " of " << k.suit() << endl;
}

File: Card/Card.py
#!/usr/bin/python

clubs = 0
spades = 1
hearts = 2
diamonds = 3


class Card:
    def __init__(self, r, s):
        self.m_card = r
        self.m_suit = s

    def value(self):
        return self.m_card

    def suit(self):
        return self.m_suit


class BlackJackCard(Card):
    def value(self):
        if self.m_card == 1:
            return 11
        elif self.m_card < 10:
            return self.m_card
        else:
            return 10

c = Card(5, clubs)
print "c is the", c.value(), "of", c.suit()
j = BlackJackCard(12, hearts)
print "j is the", j.value(), "of", j.suit()
k = BlackJackCard(1, hearts)
print "k is the", k.value(), "of", k.suit()
